integration-cli/docker_utils.go

```diff
@@ -396,6 +396,55 @@ func deleteAllContainers() error {
 	return nil
 }
 
+var protectedImages = map[string]struct{}{}
+
+func init() {
+	out, err := exec.Command(dockerBinary, "images").CombinedOutput()
+	if err != nil {
+		panic(err)
+	}
+	lines := strings.Split(string(out), "\n")[1:]
+	for _, l := range lines {
+		if l == "" {
+			continue
+		}
+		fields := strings.Fields(l)
+		imgTag := fields[0] + ":" + fields[1]
+		protectedImages[imgTag] = struct{}{}
```

Comment:
We don't, but the daemon we're testing against might have some already.
I'm just bringing up that this code as written will preserve ALL dangling
images if there are any when the tests start, and wondering if that's the
behavior we want, or if we want to consider all dangling images to be
garbage collection candidates and thus add a "<none>:<none>" special case
here to ignore them and have them be deleted.